### ISSUE TITLE
Unable To Place Blocks on Views Page Display

### ISSUE THREAD

**Issue by bobchristenson** [backdrop/backdrop-issues#4154]
If I'm understanding things correctly, it's not currently possible to place blocks on Views Page displays which have a specific (non-argument-based) URL.  That's because there's no way to create a Layout for a Views Page display.

To replicate what I'm seeing:

1.  Create a Views Page display (not a views block or any other type...an actual Views Page with a url like /myview)

2.  In order to place a block specifically on that page, you need to create a layout for it (there's no other way to place blocks on that page, correct?)

3.  The problem is, you can't setup any kind of path pattern, in the layout, which allows this.  Entering the url of the View into the 'path' section would override the views page completely.  There's no argument-based path you can put in (such as you can with node/%) in order to use a Visibility Condition and place blocks on that view page.

Basically, the way I see it, there's no way to place blocks on any page that's a Views Page.

**Comment by BWPanda** [comment]
@bobchristenson I was able to do this:

1. Create a View with a Page display with URL: `test`
1. Access `/test` and confirm view displays as expected
1. Create a Layout with path: `test`
1. Make sure the 'Main page content' block exists (if not, add it)
1. Access `/test`, clear caches and confirm view still displays as expected

At first I thought it didn't work like you were saying, but point 4 was what got it working for me.

**Comment by laryn** [comment]
> In order to place a block specifically on that page, you need to create a layout for it (there's no other way to place blocks on that page, correct?)

If it is just a single block on a single views page, you can add a block to the default layout with a visibility condition for the URL path. (If it's a lot of blocks or a lot of views this could get messy on the default layout).

But it sounds like @BWPanda has verified a cleaner solution, pending verification on your end, this is just to answer the question posed about other ways to place blocks.

**Comment by herbdool** [comment]
I can confirm @BWPanda approach.

It works though but still confusing and not sure if it could cause problems. The View has created a route (e.g. "my-content") and the new layout also has a route (e.g. "my-content"). Presumably then, we could disable the View and the route would still work but without the Views content.

**Comment by herbdool** [comment]
Something interesting. I created a View then a Layout with the same path. The Layout automatically included the Main Page Block. But if I create a Layout with a unique path unrelated to any View then there is no Main Page Block.

So maybe the Layout already checks that context?

**Comment by oadaeh** [comment]
Maybe the solution needs to be documented on a page somewhere under https://backdropcms.org/user-guide ?

**Comment by bobchristenson** [comment]
Just confirming that this all seems to work as it should.  I agree that the question is about whether the system is creating a duplicate route or not...or whether it's smart enough to deal with the fact that Views already created it.  Not sure...either way, it works as @BWPanda documented above.  Thanks!
